text/0000-sawtooth-governance.md

```diff
@@ -0,0 +1,348 @@
+or it's clear that the design has fundamental downsides that need to be
+weighed.
+
+At that point, the shepherd will announce that the RFC is in a "final comment
+period" (which lasts for one week). This is a kind of "last call" for strong
+objections to the RFC. **The announcement of the final comment period for an
+RFC should be very visible**; it should be included in the subteam's periodic
+communications.
+
+After the final comment period, the subteam can make a decision on the RFC. The
+role of the subteam at that point is *not* to reveal any new technical issues
+or arguments; if these come up during discussion, they should be added as
+comments to the RFC, and it should undergo another final comment period.
+
+Instead, the subteam decision is based on **weighing the already-revealed
+tradeoffs against the project's priorities and values** (which the root team is
+responsible for setting, globally). In the end, these decisions are about how
+to weight tradeoffs. The decision should be communicated in these terms,
+pointing out the tradeoffs that were raised and explaining how they were
+weighted, and **never introducing new arguments**.
+
+### Initial Subteams
+
+The following is a proposed list of initial subteams which is subject to change
+at the discretion of the root team, once it has formed.
+
+**Core** - Responsible for the design, architecture, performance, stability,
+ and security of the core Sawtooth platform
+
+**Application SDKs** - Responsible for the design and consistency of
+user-facing SDKs for application development
+
+**Consensus** - Responsible for the supported consensus algorithms and
+the consensus interface
+
+**Seth** - Responsible for the Sawtooth integration with Ethereum
+
+**Community Outreach** - Responsible for writing and hosting documentation,
+managing the main Sawtooth website, promoting Sawtooth to the public, managing
+demos, and providing the community with training and support
+
+**Release Management** - Responsible for release related issues such as
+dependency management, license compliance, version control, and upgrade and
+backwards compatibility
+
+**Continuous Integration** - Responsible for test and build environments and
+deployment artifacts
+
+## Contributor Permission Levels
+
+The Sawtooth community encourages contributions to the project from all
+interested individuals. The project defines three levels of permissions that
+determine what actions a contributor can perform on a particular repository.
+
+A **reviewer** has "read" permission, meaning they can review pull requests.
+
+A **committer** has "write" permission, meaning they can merge pull requests
+once they have been approved.
+
+Any community member can be promoted to a reviewer or committer by a
+maintainer. Reviewers and committers lose their status when it is removed by a
+maintainer. This is usually due to inactivity, but can also be used for
+moderation.
```

Comment:
I'm not clear what exactly the point of the reviewer/committer/maintainer split is. It seems that a committer is basically a reviewer who is able to click the 'merge' button on github, but only once two maintainers have approved the PR. I think this is getting overly complicated, and really you only need a 'committer' role with write access privileges, plus the rule that merges can only occur once two committers have approved the change? Why would there be a special class of people trusted to review PRs but not trusted to approve or merge them? This sounds like they are just ordinary GitHub users, since the repos are public, so we don't need to specify their rights here.

Reply:
I tend to agree with @grkvlt.
Maybe there's a mix here of what constraints the github permissions allow and what permissions and roles we think are necessary.
I'd like to invite any specific user to review a PR. This seems to be a limitation on github atm.
I'd like there to be a well defined limited set of people who can merge. Those individuals have the judgement to get two diverse approvals and know when it's ok to click merge. I don't care if you call them contributors with write permissions or repo maintainers, but it really seems like just one role. Either you have write perms or you don't. 
The root group maybe defines a more classical sense of 'maintainer', but again I'm not overly concerned with the labels.

Reply:
There are three distinct permissions here - review, merge, and approval. These three levels are defined here to reflect those three levels.  Review and merge is controlled via github permissions and approvers are probably in MAINTAINERS.md.  If commiters == maintainers in practice, great.  However, I suspect with more repos, we will have maintainers of some repos as committers (but not maintainers) for other repos.